pyfakefs/fake_filesystem.py

```diff
@@ -1245,6 +1250,49 @@ def CreateLink(self, file_path, link_target):
     return self.CreateFile(resolved_file_path, st_mode=stat.S_IFLNK | PERM_DEF,
                            contents=link_target)
 
+  def CreateHardLink(self, old_path, new_path):
+    """Create a hard link at new_path, pointing at old_path.
+
+    Args:
+      old_path: an existing link to the target file
+      new_path: the destination path to create a new link at
+
+    Returns:
+      the FakeFile object referred to by old_path
+
+    Raises:
+      OSError:  if something already exists at new_path
+      OSError:  if the parent directory doesn't exist
+    """
+    new_path_normalized = self.NormalizePath(new_path)
+    if self.Exists(new_path_normalized):
+      raise IOError(errno.EEXIST,
+                    'File already exists in fake filesystem',
+                    new_path)
+
+    new_parent_directory, new_basename = self.SplitPath(new_path_normalized)
+    if not new_parent_directory:
+      new_parent_directory = self.cwd
+
+    if not self.Exists(new_parent_directory):
+      raise OSError(errno.ENOENT, 'No such fake directory',
+                    new_parent_directory)
+
+    # Retrieve the target file
+    try:
+      old_file = self.GetObject(old_path)
+    except:
+      raise OSError(errno.ENOENT,
+                    'No such file or directory in fake filesystem',
+                    old_path)
+
+    old_file.st_nlink += 1
+
+    # abuse the name field to control the filename of the newly created link
+    old_file.name = new_basename
+    self.AddObject(new_parent_directory, old_file)
```

Comment:
Right, the FS size stuff wasn't around when I first implemented this. I noticed it when I rebased, there was a merge conflict in the unlink implementation, but I didn't do anything beyond just resolving that conflict. It's probably not critical to fix this issue prior to merging, though, given that both the file sizing and hard link support are new.

You can also play with these commits before they're merged, if you add a line like this into your git config:
```
[remote "origin"]
...
	fetch = +refs/pull/*/head:refs/remotes/origin/pr/*
```

That will cause git fetch to make all open PRs available under origin/pr/<number>.

It would make sense for me to update this myself, but I've been pretty busy in the last few weeks, and I'm not sure when I'd find time to get it done.